android/src/main/java/com/amplitude/android/Configuration.kt

```diff
@@ -46,6 +46,7 @@ open class Configuration @JvmOverloads constructor(
     override var identifyInterceptStorageProvider: StorageProvider = AndroidStorageProvider(),
     override var identityStorageProvider: IdentityStorageProvider = FileIdentityStorageProvider(),
     var migrateLegacyData: Boolean = true,
+    var deviceId: String? = null,
```

Comment:
If we add deviceId to the configuration we should probably keep it up to date with the current value.